### PR TITLE
Added pvs parameter to lvol module

### LINKED ISSUE #2003: Add pvs option to lvol module
##### ISSUE TYPE

- Feature Idea
##### COMPONENT NAME

lvol
##### ANSIBLE VERSION

```
<!--- Paste verbatim output from “ansible --version” between quotes -->
```
##### SUMMARY

Add pvs option, where we could list physical volumes which should be used in lvol.
Exapmle: LV CACHE volumes must be in same vg as ORIGIN LV's for which they serve as cache,
apparently we want to have SSD disks for LV CACHE and HDD for ORIGIN LV's, we don't want to mix them up in one LV
##### STEPS TO REPRODUCE

```
<!--- Paste example playbooks or commands between quotes -->
```

##### ACTUAL RESULTS

```
<!--- Paste verbatim command output between quotes -->
```

### PR DESCRIPTION
**Pull request by p53**
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

lvol
##### ANSIBLE VERSION

ansible 2.0.0.2
##### SUMMARY

Adding pvs parameter to lvol module. We need it if we want to assign different disks to
different lv's. E.g. in case we are using cached lvol's

ok: [127.0.0.1] => (item={u'vg': u'vgpool', u'name': u'lvcacheswap', u'opts': u'', u'size': u'100M'})

ok: [127.0.0.1] => (item={u'pvs': u'/dev/sdb', u'vg': u'vgpool', u'name': u'lvcacheswap', u'opts': u'', u'size': u'100M'})

Fixes #2003

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @p53. To the current maintainers, @jhoekx, @abulimov please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: system/lvol.py [hunk 1]
```diff
@@ -119,7 +125,7 @@ def parse_lvs(data):
     for line in data.splitlines():
         parts = line.strip().split(';')
         lvs.append({
-            'name': parts[0],
+            'name': parts[0].replace('[','').replace(']',''),
```
Comment (abulimov) [inline system/lvol.py:133]: Could you please explain why do we need to remove [] letters? I've never saw them in lvs output.

Comment (p53) [inline system/lvol.py:133]: because cache pool lvols are by default hidden, you must show them with lvs -a and in the output they will show up like this [some_lv], try to create cached lv's see man lvmcache, this will at least enable possibility to create them with your module. i am preparing separate module for managing cached lv's because it is not so easy as normal lv , you cannot resize them while they are cached...

**Comment by abulimov** [comment]
Thanks, shipit!

**Comment by gregdek** [comment]
Thanks again to @p53 for this PR, and thanks @jhoekx, @abulimov for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by p53** [comment]
Thanks! if you would like to try out lvol_cache module i made pull request #2041

**Comment by p53** [comment]
Thanks! btw i created new lvol_cache module, if you like you can try it out #2041

File: system/lvol.py [hunk 2]
```diff
@@ -181,6 +188,12 @@ def main():
     size_opt = 'L'
     size_unit = 'm'
     snapshot = module.params['snapshot']
+    pvs = module.params['pvs']
```
Comment (resmo) [inline system/lvol.py]: pvs param is missing in the doc

**Comment by resmo** [comment]
needs_revision

File: system/lvol.py [hunk 3]
```diff
@@ -76,6 +81,12 @@
 # Create a logical volume of 512m.
 - lvol: vg=firefly lv=test size=512
 
+# Create a logical volume of 512m with disks /dev/sda and /dev/sdb
+- lvol: vg=firefly lv=test size=512 pvs=/dev/sda,/dev/sdb
```
Comment (resmo) [inline system/lvol.py:85]: `pvs` param is `type='str'` but it should rather be `type='list'`. You will get a python list returned, but you can pass the values as shown above in the example

```
lvol: vg=firefly lv=test size=512 pvs=/dev/sda,/dev/sdb
```

but also as list

```
lvol: 
  vg: firefly
  lv: test
  size: 512
  pvs:
    - /dev/sda
    - /dev/sdb
```

Of course you will need to adjust the code to handle it as python list. e.g. Line 201 `pvs = ' '.join(pvs)`

File: system/lvol.py [hunk 4]
```diff
@@ -230,12 +248,12 @@ def main():
...
 
     vgs = parse_vgs(current_vgs)
-    this_vg = vgs[0]   
+    this_vg = vgs[0]
 
     # Get information on logical volume requested
     lvs_cmd = module.get_bin_path("lvs", required=True)
     rc, current_lvs, err = module.run_command(
-        "%s --noheadings --nosuffix -o lv_name,size --units %s --separator ';' %s" % (lvs_cmd, unit, vg))
+        "%s -a --noheadings --nosuffix -o lv_name,size --units %s --separator ';' %s" % (lvs_cmd, unit, vg))
```
Comment (resmo) [inline system/lvol.py:256]: Is this related to the feature or is it a bug fix?
